### PR TITLE
Fixed display of Category tree on Product page

### PR DESCRIPTION
**Pull request by tomlev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | Fix display on product page for category tree
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-4331
| How to test?  | Go on product page on BO. Add a category and check it is correctly displayed on tree. Expand all existing categories and check th tree is still correctly displayed

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8526)

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Restarting tests, but saw in the log:

`localhost:80 127.0.0.1 - - [22/Nov/2017:14:15:38 +0000] "POST /admin-dev/index.php/product/catalog/0/20/id_product/desc?_token=2ehDseHbXAzH0l0gVZrV9P06t17GKSgGlqhqGolsPK8 HTTP/1.1" 500 796 "http://localhost/admin-dev/index.php/product/catalog/last?_token=2ehDseHbXAzH0l0gVZrV9P06t17GKSgGlqhqGolsPK8" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.113 Safari/537.36"`

File: admin-dev/themes/new-theme/public/translations.bundle.js [hunk 1]
```diff
@@ -28177,7 +28177,7 @@ var Component = __webpack_require__(3)(
   /* cssModules */
   null
 )
-Component.options.__file = "/home/bbieri/repos/PrestaShop/admin-dev/themes/new-theme/js/app/widgets/ps-button.vue"
+Component.options.__file = "/Users/thomasleviandier/Documents/projects/prestashop/PrestaShop/admin-dev/themes/new-theme/js/app/widgets/ps-button.vue"
```
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/public/translations.bundle.js]: I've never noticed we have commited this kind of information :/

Comment (tomlev) [reply admin-dev/themes/new-theme/public/translations.bundle.js]: internal usage of webpack DOES expose these informations :/
packaging should delete this, we should work on webpack configuration...

Comment (eternoendless) [reply admin-dev/themes/new-theme/public/translations.bundle.js]: Yeah, it's an issue I've been aware of for some time, it's related to webpack and (I think) hot module replacement which we need to deactivate for production builds.
I'm working on it.

**Comment by eternoendless** [comment]
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 32 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXJ2Ndin1hbC3pmAL:-KzyXJ2Ndin1hbC3pmAM:b-7655cc) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L32)):*
> ```JavaScript
>       $('.more').toggleClass('less');
>       // scroll right to see the radio buttons
>       scrollCategoryTree();
> ```

You shouldn't use functions before declaring it. If you're curious about why, read about _hoisting_ here: https://github.com/airbnb/javascript#hoisting

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 61 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyW7BX4CAzwYoZiZzp:-KzyW7BYBIZM0IYrSMIN:b-reirnz) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L61)):*
> ```JavaScript
>   });
> 
>   function scrollCategoryTree() {
> ```

This should be declared as follows:

```js
const scrollCategoryTree = function scrollCategoryTree() {
}
```

Find out more about why here: https://github.com/airbnb/javascript#functions--declarations

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 61 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXKBZrIPzMNk551JE:-KzyXKBZrIPzMNk551JF:b-pr9a4) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L61)):*
> ```JavaScript
>   });
> 
>   function scrollCategoryTree() {
> ```

This function is missing its [JSDoc header](usejsdoc.org)

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 62 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXRoOJwSqTfRvkFVE:-KzyXRoOJwSqTfRvkFVF:byoswok) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L62)):*
> ```JavaScript
> 
>   function scrollCategoryTree() {
>       var leftPos = $('.category-tree-overflow').width();
> ```

The selector `$('.category-tree-overflow')` is performed twice

---

*[admin-dev/themes/new-theme/scss/pages/_product_page.scss, line 593 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXlJDIwiBTnDOhGQ0:-KzyXlJDIwiBTnDOhGQ1:br43ptp) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/scss/pages/_product_page.scss#L593)):*
> ```SCSS
>       label {
>         position: relative;
>         padding-right: 24px;
> ```

Please use rems instead of px so that all sizes are proportional

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8526)*

**Review by eternoendless — CHANGES_REQUESTED** [review]
Some small changes required

**Comment by tomlev** [comment]
Review status: 3 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 32 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXJ2Ndin1hbC3pmAL:-L-2Wb-UdvTUHLvRWPyG:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L32)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

You shouldn't use functions before declaring it. If you're curious about why, read about _hoisting_ here: https://github.com/airbnb/javascript#hoisting
</blockquote></details>

Done.

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 61 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyW7BX4CAzwYoZiZzp:-L-2Wd6Gv3dxPGCKWStc:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L61)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

This should be declared as follows:

```js
const scrollCategoryTree = function scrollCategoryTree() {
}
```

Find out more about why here: https://github.com/airbnb/javascript#functions--declarations
</blockquote></details>

Done.

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 61 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXKBZrIPzMNk551JE:-L-2WeBc5bVyiAK-5MIe:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L61)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

This function is missing its [JSDoc header](usejsdoc.org)
</blockquote></details>

Done.

---

*[admin-dev/themes/new-theme/js/product-page/category-tree.js, line 62 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXRoOJwSqTfRvkFVE:-L-2WemaVsh8hruw79qi:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/js/product-page/category-tree.js#L62)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

The selector `$('.category-tree-overflow')` is performed twice
</blockquote></details>

Done.

---

*[admin-dev/themes/new-theme/scss/pages/_product_page.scss, line 593 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-KzyXlJDIwiBTnDOhGQ0:-L-2YKV5dN9iNHGCjxo1:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/9f4b38b3e13459eb93095dde2a7e0e694dfee605/admin-dev/themes/new-theme/scss/pages/_product_page.scss#L593)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Please use rems instead of px so that all sizes are proportional
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8526)*

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 6 of 6 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8526#-:-L-69QK_8uPyQGhMGjXz:bnfp4nl)*

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @tomlev

**Pull request merged**